process/economics/rewardsConfigHandler.go

```diff
@@ -148,26 +176,30 @@ func checkAndParseRewardsSettings(
 
 		topUpGradientPoint, _ := big.NewInt(0).SetString(rewardsCfg.TopUpGradientPoint, 10)
 
-		decodedAddress, err := pubkeyConverter.Decode(rewardsCfg.ProtocolSustainabilityAddress)
+		protocolSustainability, err := decodeAddressAndVerifyShard(pubkeyConverter, shardCoordinator, rewardsCfg.ProtocolSustainabilityAddress)
+		if err != nil {
+			return nil, err
+		}
+
+		ecosystemGrowth, err := decodeAddressAndVerifyShard(pubkeyConverter, shardCoordinator, rewardsCfg.EcosystemGrowthAddress)
 		if err != nil {
-			log.Warn("invalid protocol sustainability reward address",
-				"err", err,
-				"provided address", rewardsCfg.ProtocolSustainabilityAddress,
-				"epoch", rewardsCfg.EpochEnable,
-			)
 			return nil, err
 		}
 
-		protocolSustainabilityShardID := shardCoordinator.ComputeId(decodedAddress)
-		if protocolSustainabilityShardID == core.MetachainShardId {
-			return nil, process.ErrProtocolSustainabilityAddressInMetachain
+		growthDividend, err := decodeAddressAndVerifyShard(pubkeyConverter, shardCoordinator, rewardsCfg.GrowthDividendAddress)
+		if err != nil {
+			return nil, err
 		}
 
 		rewardsConfigSlice = append(rewardsConfigSlice, &rewardsConfig{
 			rewardsSettingEpoch:              rewardsCfg.EpochEnable,
 			leaderPercentage:                 rewardsCfg.LeaderPercentage,
 			protocolSustainabilityPercentage: rewardsCfg.ProtocolSustainabilityPercentage,
-			protocolSustainabilityAddress:    string(decodedAddress),
+			protocolSustainabilityAddress:    string(protocolSustainability),
+			ecosystemGrowthPercentage:        rewardsCfg.EcosystemGrowthPercentage,
```

Comment:
the percentages should be validated that their sum does not get over 100% or below 0%.

Reply:
done